src/Compiler/test/Microsoft.NET.Sdk.Razor.SourceGenerators.Tests/DuplicateRazorFileIncludedAnalyzerTest.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.SourceGenerators;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Testing;
+using Microsoft.CodeAnalysis.Testing;
+using Xunit;
+
+namespace Microsoft.NET.Sdk.Razor.SourceGenerators.Tests;
+
+public class DuplicateRazorFileIncludedAnalyzerTest
+{
+    [Theory]
+    [InlineData("Duplicate.cshtml")]
+    [InlineData("Duplicate.razor")]
+    public async Task Analyzer_ReportsDiagnostic_WhenDuplicateRazorFileIsIncluded(string fileName)
```

Comment:
Consider adding a test for the razor source generator as well to see what the errors would be.

I actually cannot reproduce any issues there. I've created a project which passes duplicate AdditionalFiles to Csc but don't see any errors from the razor source generator. What is needed to hit an error?